api/src/main/java/com/github/retrooper/packetevents/util/PacketTransformationUtil.java

```diff
@@ -51,6 +54,24 @@ public static PacketWrapper<?>[] transform(PacketWrapper<?> wrapper) {
                     Equipment equipment = entityEquipment.getEquipment().get(i);
                     output[i] = new WrapperPlayServerEntityEquipment(entityEquipment.getEntityId(), Collections.singletonList(equipment));
                 }
+                return output;
+            }
+        } else if (wrapper instanceof WrapperPlayServerChunkData) {
+            WrapperPlayServerChunkData chunkData = (WrapperPlayServerChunkData) wrapper;
+            LightData lightData = chunkData.getLightData();
+
+            if (chunkData.getServerVersion().isOlderThan(ServerVersion.V_1_18) && lightData != null) {
+                //Transform into chunk data & light data packets
+                PacketWrapper<?>[] output = new PacketWrapper[2];
+                output[0] = new WrapperPlayServerUpdateLight(
+                        chunkData.getColumn().getX(),
+                        chunkData.getColumn().getZ(),
+                        lightData
+                );
+                // Light packet must be sent first
+                output[1] = chunkData;
+                chunkData.setLightData(null);
```

Comment:
should light data really be deleted upon transforming packets?
this gets called each packet sending, so sending this one packet wrapper to multiple players will only send light data to the first one as it gets deleted here

Reply:
make sense!